src/Common/Bridge/BridgeFacadeMethodsRule.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace ArchitectureSniffer\Common\Bridge;
+
+use ArchitectureSniffer\SprykerAbstractRule;
+use PHPMD\AbstractNode;
+use PHPMD\Node\ClassNode;
+use PHPMD\Node\MethodNode;
+use PHPMD\Rule\ClassAware;
+use ReflectionMethod;
+use ReflectionType;
+
+class BridgeFacadeMethodsRule extends SprykerAbstractRule implements ClassAware
+{
+    /**
+     * @var string
+     */
+    protected const RULE = 'Based on naming convention the bridge facade method must follow CRUD signature.';
+
+    /**
+     * @return string
+     */
+    public function getDescription()
+    {
+        return static::RULE;
+    }
+
+    /**
+     * @param \PHPMD\AbstractNode $node
+     *
+     * @return void
+     */
+    public function apply(AbstractNode $node)
+    {
+        if (
+            preg_match('([A-Za-z0-9]+Bridge$)', $node->getName()) === 0 ||
+            preg_match('#.*\\\\Dependency\\\\Facade.*#', $node->getNamespaceName()) === 0 ||
+            !$node instanceof ClassNode
+        ) {
+            return;
+        }
+        $bridgeName = $node->getFullQualifiedName();
+
+        foreach ($node->getMethods() as $method) {
+            $interfaceMethodName = sprintf('%s::%s', $bridgeName, $method->getName());
+            $interfaceMethodReflection = new ReflectionMethod($interfaceMethodName);
+            $methodReturnType = $interfaceMethodReflection->getReturnType();
+
+            if (preg_match('/^(get|read|find)/', $method->getName())) {
+                $this->verifyGetMethod($method, $methodReturnType);
+
+                continue;
+            }
+
+            if (preg_match('/^(delete|remove)/', $method->getName())) {
+                $this->verifyDeleteMethod($method, $methodReturnType);
+
+                continue;
+            }
+
+            if (preg_match('/^(create|add)/', $method->getName())) {
+                $this->verifyCreateMethod($method, $methodReturnType);
+
+                continue;
+            }
+
+            if (preg_match('/^(update|change)/', $method->getName())) {
+                $this->verifyUpdateMethod($method, $methodReturnType);
+
+                continue;
+            }
+
+            if (strpos($method->getName(), 'save') === 0) {
+                $this->addViolation($method, [sprintf(
+                    'Method `%s()` must have `public function update(create)<DomainEntity>Collection(<DomainEntity>CollectionRequestTransfer): <DomainEntity>CollectionResponseTransfer;` signature.',
```

Comment:
```suggestion
                    'Method `%s()` must have `public function [update|create]<DomainEntity>Collection(<DomainEntity>CollectionRequestTransfer): <DomainEntity>CollectionResponseTransfer;` signature.',
```
Would this be maybe more clear?